### DISCUSSION TITLE
Wide Color - problem with the color red

### DISCUSSION THREAD

**Discussion by Pax709** [gopro/labs#476] [Q&A]
Hello, I would like to know more about the new feature in GOPRO LABS (HERO 11) wide color, it seems to me that there is a problem with the red color, it is too orange and trying to correct it shifts the skin tones in the wrong direction, the usual FLAT/Rec709 is very well calibrated in this area.

When Gopro creates Rec709, does it access the full raw data from the matrix before writing to HEVC and therefore the red color and skin is very good?

Can we assume , that the wide color in this form what was given to us(H. 265), will have problems with red color and trying to correct it is doomed to failure?

I will write right away, I have a professional wide gamut monitor, I checked the colors eyeball to eyeball with the monitor and the wide color has a red problem.

I would appreciate an explanation of what this wide color is all about.

**Comment by dnewman-gpsw** [comment]
WIDE color, is the native gamut of the sensor, it is significantly wider the Rec709 gamut, but it does leave more work for the editor/colorist.  However, there is no inherent problem, as the Rec709 you see came from the wide gamut image (using only a 3x3 color matrix).  So you just need a tool like https://mattroberts.org/MBR_Color_Corrector/ to help you build that matrix/LUT externally (or to target a other wider gamuts.)

**Comment by dnewman-gpsw** [comment]
Based on the images you sent (note: images can be attached here).  Are you running the camera 10-bit?  Log 400 and WIDE is not suitable for 8-bit encoding.  I'm seeing macro-block artifacts in the color, this is sign that you are either encoding at 8-bit, or you are not running sufficiently high enough bit-rate.

**Comment by Pax709** [comment]
Exported frames in with Davinci Resolve - TIFF 48bit, then took screenshot.

![Screenshot](https://github.com/gopro/labs/assets/141012631/5048ac83-b112-40ca-8f81-36a9cde82a43)
![Screen](https://github.com/gopro/labs/assets/141012631/371f43eb-e11e-48c0-a5ff-c71265856c26)

Original frame in TIFF and attached video file metadata, obviously recorded in 10 bit with Bitrate 170.

https://drive.google.com/drive/folders/1mhOah-tfUq-dJKkeOYHJSIZxY_bs-2gf

**Reply by dnewman-gpsw** [reply]
I can't confirm the 10-bit from the metadata, that requires the MP4. You can verify the 10-bit using ffprobe, or your NLE might say also.  Premiere Pro properties lists **Video Codec Type: HEVC 10 bit  4:2:0  (Full Range)** for my HERO11 media.  There is clearly chroma macro-blocking issues, that I can't produce with my own cameras in 10-bit (even a little surprising for 8-bit.)  Even at 120Mb/s the images aren't doing that is my our tests.  Try testing NR01=50 (or 1 for more noise+detail), this might help the chroma compression encode more details.

**Reply by dnewman-gpsw** [reply]
"This effect is after applying the color board I bought"  - I don't follow this. 

The issues is color macro-blocking, not the LUT.  So I would try difficult decoders, just in case DaVinci is doing something weird (some hard decoders aren't great,) but it seems to be in the source, potentially the camera.  Try shooting a similar scene that you can share, add NR01=1 to see if that helps.

**Reply by Pax709** [reply]
Well no, I have the same two files in the timeline :
1.In the first one I change the colors and I don't get this nasty effect.

2.In the second file I apply a supposedly professional Lut and get this effect immediately.

So why do you assume that it is not LUT ?

**Reply by dnewman-gpsw** [reply]
I can see the error without the LUT, plus LUTs can't make squares. The LUT is fine

**Comment by Pax709** [comment]
I've attached the original file, I can't send you LeemingLut because it will be illegal, but you could show what you can get from this file using the right color table ?

https://drive.google.com/drive/folders/1mhOah-tfUq-dJKkeOYHJSIZxY_bs-2gf

**Reply by dnewman-gpsw** [reply]
Still uploading? As that link is only the .tiff and metadata.

**Reply by Pax709** [reply]
I can send you the original video file, but the face can not be revealed, where to send You the link ?

**Reply by dnewman-gpsw** [reply]
Check your email.

**Comment by dnewman-gpsw** [comment]
While this discussion goes direct.  There doesn't appear to LUT or a camera problem, mostly like a system configuration issue. In decoding the source on an other system the chroma artifacts disappeared. We will follow up once we determine this cause and workarounds if this occurs for anyone else.

**Comment by Pax709** [comment]
At this stage, the main problem has been solved.

I have a question about the wide color ( wide=1), I would like to find out what the smoothness of the color is due to? , comparing with rec709 there is a very big difference, is wide color does not use so-called dithering?. The Rec709 shows noise, the color is not smooth.

**Reply by dnewman-gpsw** [reply]
The camera does not use dithering, other than in its natural form - noise. WIDE reduces the magnitude of chroma, so that larger chroma range can be supported.  Saturating the image in camera, magnifies noise a little, it is like turning up the contrast. There is a second process that is likely contributing, the camera will denoise low threshold noise, and when the rec709 saturation is not applied, chroma signal is lower, more denoising might be applied. This one of the reasons I suggested you try using !MNR01=50 (or lower), as this will decrease the noise reduction. This gives more signal detail, and noise, across all chroma channels.  Using NR01 gives you a better image to denoise in post. Decreasing NR01, should be accompanied with a moderate increase in BITR. The camera is tuned for normal uses, when you modify it operations with WIDE or similar some other tuning will be desirable (i.e. more signal with higher bitrates to help in post.)

**Reply by Pax709** [reply]
The topic can be considered solved, any problems are due to 3 reasons :

1. H265 and chrominance macro blocking - the solution is to encode to ProRes - Shutterencoder.
2. Using GP11 "WIDE" profile, cut the beginning of the recording about 2 seconds, maybe GOPRO will fix it....
3. LUT - is to be superimposed in the second node ( first - WB) and not the last.

Thanks David for your help ;)

**Reply by dnewman-gpsw** [reply]
For clarity: 
1. If your PC's hardware issue can't decode at 10-bit for 5.3K, use a tool like Shutterencoder to transcode before editing.
2. Not really a bug to fix, working as designed. Encoders in camera's ramp quality to match the target bitrate, this taken 15-30 frames to do so. It is the nature of long GOP compression.

**Reply by Pax709** [reply]
Then I'll ask why this doesn't occur in Rec709, there is a different shorter compression there?

**Reply by dnewman-gpsw** [reply]
It occurs there and in every mode, yet in Rec709 the color has more signal, so it is more likely to survive heavy compression. WIDE has less chroma signal so it can encode wider color, that is the only change. Compression has to guess at the first frame quantization level, it guesses for more than average compression, as to guess wrong only has minor consequences. If it did the reverse (quantize too little) captures can fail as there could be too much data to store. This is classic long-GOP rate control. Compress the first frame, look at the resulting bit-rate, and adjust quantization as a prediction for the next frame, repeat until the target bit-rate or quality is achieved. All cameras need to do this to be performant, as to re-encode frames is too costly on battery and thermally.
